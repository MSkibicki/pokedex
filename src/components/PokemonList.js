import React, { useEffect, useState } from "react";
import axios from "axios";
import SinglePokemon from "./SinglePokemon";

const App = () => {
  const [pokemons, setPokemons] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const getData = async () => {
      const result = await axios(`https://pokeapi.co/api/v2/pokemon`);

      setLoading(false);

      setPokemons(result.data["results"]);
    };

    getData();
  }, []);

  
  if (loading) return "Loading...";

  return (
    <>
      {pokemons.map((pokemon) => (
        <SinglePokemon
          key={pokemon.name}
          name={pokemon.name}
          url={pokemon.url}
        />
      ))}
    </>
  );
};

export default App;
