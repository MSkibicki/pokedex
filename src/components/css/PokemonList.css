/* No CSS *//*# sourceMappingURL=PokemonList.css.map */