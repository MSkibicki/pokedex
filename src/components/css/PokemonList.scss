.pokemon {
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
}
.loading {
  text-align: center;
  margin-top: 10em;
  color: #f2f7eb;
  text-shadow: 2px 2px 2px black;
}
