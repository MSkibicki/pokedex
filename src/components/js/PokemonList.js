import React, { useEffect, useState } from "react";
import axios from "axios";
import SinglePokemon from "./SinglePokemon";
import Pagination from "./Pagination";
import PokemonFilter from "./PokemonFilter";

const App = () => {
  const [pokemons, setPokemons] = useState([]);
  const [loading, setLoading] = useState(false);
  const [currentPage, setCurrentPage] = useState(
    "https://pokeapi.co/api/v2/pokemon"
  );
  const [previousPage, setPreviousPage] = useState();
  const [nextPage, setNextPage] = useState();
  const [inputValue, setInputValue] = useState("");

  useEffect(() => {
    const getData = async () => {
      setLoading(true);

      let token;
      const result = await axios(currentPage, {
        cancelToken: new axios.CancelToken((c) => (token = c)),
      });

      setLoading(false);
      setPreviousPage(result.data.previous);
      setNextPage(result.data.next);
      setPokemons(result.data.results);

      return () => token.cancel();
    };

    getData();
  }, [currentPage]);

  const handlePreviousPage = () => {
    setCurrentPage(previousPage);
  };

  const handleNextPage = () => {
    setCurrentPage(nextPage);
  };

  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  let filterPokemons = pokemons.filter((pokemon) => {
    return pokemon.name.toLowerCase().includes(inputValue.toLowerCase());
  });

  if (loading) return "Loading...";

  return (
    <>
      <PokemonFilter handleInput={handleInput} />
      {filterPokemons.map((filteredPokemon) => (
        <SinglePokemon
          key={filteredPokemon.name}
          name={filteredPokemon.name}
          url={filteredPokemon.url}
        />
      ))}
      <Pagination
        handlePreviousPage={previousPage ? handlePreviousPage : null}
        handleNextPage={nextPage ? handleNextPage : null}
      />
    </>
  );
};

export default App;
