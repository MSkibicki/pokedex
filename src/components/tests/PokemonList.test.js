import PokemonList from "./../js/PokemonList";

it("renders correctly", () => {
  const wrapper = shallow(<PokemonList />);

  expect(wrapper).toMatchSnapshot();
});
