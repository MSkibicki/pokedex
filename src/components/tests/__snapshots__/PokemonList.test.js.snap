// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Navbar />
  <div
    className="pokemon-list"
  >
    <PokemonFilter
      handleInput={[Function]}
    />
    <div
      className="pokemon"
    />
    <Pagination
      handleNextPage={null}
      handlePreviousPage={null}
    />
  </div>
</Fragment>
`;
